xxx-package-module-property.md

```diff
@@ -0,0 +1,273 @@
+| Title  | Package.json Module Property |
+|--------|------------------------------|
+| Author | @guybedford                  |
+| Status | DRAFT                        |
+| Date   | 2017-07-13                   |
+
+## 1. Background
+
+This proposal specifies the `"module"` property in the package.json, building
+on the previous work in the
+[In Defense of Dot JS](https://github.com/dherman/defense-of-dot-js/blob/master/proposal.md)
+proposal (DDJS), as well as many other discussions.
+
+Instead of supporting the additional `"modules"` and `"modules.root"`
+properties from that proposal, this proposal aims to adjust the handling of
+`"module"` slightly so that it is the only property supported.
+
+A draft specification of the NodeJS module resolution algorithm with this
+adjustment is included in section 5. Draft Specification.
+
+## 2. Motivation
+
+There is still uncertainty as to how exactly to distinguish an ES Module from
+a CommonJS module. While `.mjs` and `"use module"`
+(https://github.com/tc39/proposal-modules-pragma) act as useful indicators,
+these properties act as file-specific indicators of the module format. If we
+are to keep the `.js` extension without making `"use module"` mandatory, then
+there is also a need for a more global indication that a package contains only
+ES modules.
```

Comment:
That's a key question certainly, and a large part of the discussion in this thread.